src/Symfony/Component/HttpFoundation/RequestMatcher/HeaderRequestMatcher.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation\RequestMatcher;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestMatcherInterface;
+
+/**
+ * Checks the HTTP headers of a Request.
```

Comment:
Can we be more precise here? After reading the code, I understood that the class matches request where a given header is simply present. We should point out that we cannot match on the value of headers with this class.